presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergLogicalPlanner.java

```diff
@@ -240,6 +247,836 @@ public void testFiltersWithPushdownDisable()
         assertUpdate("DROP TABLE test_filters_with_pushdown_disable");
     }
 
+    @DataProvider(name = "timezones")
+    public Object[][] timezones()
+    {
+        return new Object[][] {
+                {"UTC", true},
+                {"America/Los_Angeles", true},
+                {"Asia/Shanghai", true},
+                {"None", false}};
+    }
+
+    @Test(dataProvider = "timezones")
+    public void testHourTransform(String zoneId, boolean legacyTimestamp)
+    {
+        SessionBuilder sessionBuilder = Session.builder(getSession())
+                .setSystemProperty(LEGACY_TIMESTAMP, String.valueOf(legacyTimestamp));
+        if (legacyTimestamp) {
+            sessionBuilder.setTimeZoneKey(TimeZoneKey.getTimeZoneKey(zoneId));
+        }
+        Session session = sessionBuilder.build();
+        assertUpdate(session, "CREATE TABLE test_hour_transform (d TIMESTAMP, b BIGINT) WITH (partitioning = ARRAY['hour(d)'])");
+
+        assertPlan(session, "SELECT * FROM test_hour_transform WHERE d IS NOT NULL",
+                output(exchange(strictTableScan("test_hour_transform", identityMap("d", "b")))));
+        assertPlan(session, "SELECT * FROM test_hour_transform WHERE d IS NULL",
+                output(exchange(strictTableScan("test_hour_transform", identityMap("d", "b")))));
```

Comment:
nit: If I'm parsing these tests right, I think you are really just checking that the table scan has the correct parameters. We don't care whether the output or exchanges or there. Should be able to replace those with `anyTree` instead. This goes for almost all of the added tests

```suggestion
anyTree(strictTableScan("test_hour_transform", identityMap("d", "b")));
```

Reply:
I think the main purpose here is to check whether there exist a `FilterNode` after the pushdown. And `anyTreeI()` would bring in ambiguity.  what's your opinion, is there a better way to demonstrate something like `here do not exist FilterNode`?

Reply:
You could try `anyNot` from `PlanMatchPattern`. I think this will also make the intentions more clear.

```suggestion
                anyNot(FilterNode.class, anyTree(strictTableScan("test_hour_transform", identityMap("d", "b"))));
```

Reply:
Thank you for the information, I will try it.